libs/auth/src/user.ts

```diff
@@ -0,0 +1,75 @@
+import { prisma } from '@nmit-coursition/db'
+import type { cas__user } from '@prisma/client'
+
+interface User {
+  id: string
+  email: string
+  profilePictureUrl?: string
+  firstName?: string
+  lastName?: string
+  createdAt: string
+  updatedAt: string
+}
+
+export async function updateUser(userData: User, organisationId: number): Promise<cas__user> {
+  const user =
+    (await prisma.cas__user.findFirst({
+      where: {
+        AND: [{ OR: [{ workos_id: userData.id }, { email: userData.email }] }, { organisation_id: organisationId }],
+      },
+    })) ||
+    (await prisma.cas__user.create({
+      data: {
+        organisation_id: organisationId,
+        email: userData.email,
+        inserted_date: new Date(userData.createdAt),
+        updated_date: new Date(userData.updatedAt),
+        synced_date: new Date(),
+        workos_id: userData.id,
+      },
+    }))
+
+  return await prisma.cas__user.update({
```

Comment:
**suggestion:** We shouldn't await in return, it's already a promise, we can just return it without awaiting.